ocaml/sdk-gen/component-test/spec/session.json

```diff
@@ -36,5 +36,665 @@
+      "session.logout": ["login successfully"]
+    },
+    "expected_result": {
+      "session.login_with_password": {
+        "result": "login successfully"
+      },
+      "pool.get_all": {
+        "result": ["poolref0"]
+      },
+      "pool.get_record": {
+        "result": {"name_label": "pool0"}
+      },
+      "host.get_record": {
+        "result": {
+          "name_label": "host0",
+          "software_version": {"xapi": "1.20"}
+        }
+      }, 
+      "VM.query_data_source": {
+        "result": "1.0"
+      },
+      "session.logout": {
+        "result": "logout successfully"
+      }
+    }
+  },
+  "test_id11": {
+    "method": [
+      "session.login_with_password", 
+      "pool.get_all", 
+      "pool.get_record",
+      "host.get_record",
+      "VM.set_shadow_multiplier_live",
+      "session.logout"
+    ],
+    "params": {
+      "session.login_with_password": ["", "", "1.0", "Go sdk component test"],
+      "pool.get_all":  ["login successfully"],
+      "pool.get_record": ["login successfully", "poolref0"],
+      "host.get_record": ["login successfully", ""],
+      "VM.set_shadow_multiplier_live": ["login successfully", "vmref0", 2.0],
+      "session.logout": ["login successfully"]
+    },
+    "expected_result": {
+      "session.login_with_password": {
+        "result": "login successfully"
+      },
+      "pool.get_all": {
+        "result": ["poolref0"]
+      },
+      "pool.get_record": {
+        "result": {"name_label": "pool0"}
+      },
+      "host.get_record": {
+        "result": {
+          "name_label": "host0",
+          "software_version": {"xapi": "1.20"}
+        }
+      }, 
+      "VM.set_shadow_multiplier_live": {
+        "result": ""
+      },
+      "session.logout": {
+        "result": "logout successfully"
+      }
+    }
+  },
+  "test_id12": {
+    "method": [
+      "session.login_with_password", 
+      "pool.get_all", 
+      "pool.get_record",
+      "host.get_record",
+      "VM.get_record",
+      "session.logout"
+    ],
+    "params": {
+      "session.login_with_password": ["", "", "1.0", "Go sdk component test"],
+      "pool.get_all":  ["login successfully"],
+      "pool.get_record": ["login successfully", "poolref0"],
+      "host.get_record": ["login successfully", ""],
+      "VM.get_record": ["login successfully", "vmref0"],
+      "session.logout": ["login successfully"]
+    },
+    "expected_result": {
+      "session.login_with_password": {
+        "result": "login successfully"
+      },
+      "pool.get_all": {
+        "result": ["poolref0"]
+      },
+      "pool.get_record": {
+        "result": {"name_label": "pool0"}
+      },
+      "host.get_record": {
+        "result": {
+          "name_label": "host0",
+          "software_version": {"xapi": "1.20"}
+        }
+      }, 
+      "VM.get_record": {
+        "result": { 
+          "name_label": "vm0"
+        },
+        "error": {
+          "code": "abcd",
+          "message": 123456,
```

Comment:
The "code" format error as expected. But the "message" here should be "string"

Reply:
This is for testing Rpc Error, so that it will return the error "cannot unmarshal string into Go struct field ResponseError.error.code of type int"